tests/test_cssselect.py

```diff
@@ -453,6 +472,24 @@ def xpath(css):
         assert xpath("e:EmPTY") == ("e[not(*) and not(string-length())]")
         assert xpath("e:root") == ("e[not(parent::*)]")
         assert xpath("e:hover") == ("e[0]")  # never matches
+        assert xpath("*:not(a > b)") == (
+            "*[not([a] and parent::*[b])]"
+        )  # select anything that is not b or doesn't have a parent a
+        assert xpath("*:not(a + b)") == (
+            "*[not([a] and following-sibling::*[position()=1 and b])]"
+        )  # select anything that is not b or doesn't have an immediate sibling a
+        assert xpath("*:not(a ~ b)") == (
+            "*[not([a] and following-sibling::*[b])]"
+        )  # select anything that is not b or doesn't have a sibling a
+        assert xpath("*:not(a b)") == (
+            '*[not(name()="b" and ancestor::*[name()="a"])]'
```

Comment:
This change should also be applied to other 3 combinators.

Reply:
So I suppose the tests should look like this:
```
assert xpath("*:not(a > b)") == ('*[not(name()="b" and parent::*[name()="a"])]')  # select anything that is not b or doesn't have a parent a
assert xpath("*:not(a + b)") == ('*[not(name()="b" and following-sibling::*[position()=1 and name()="a"])]' )  # select anything that is not b or doesn't have an immediate sibling a
assert xpath("*:not(a ~ b)") == ('*[not(name()="b" and following-sibling::*[name()="a"])]')  # select anything that is not b or doesn't have a sibling a
assert xpath("*:not(a b)") == ('*[not(name()="b" and ancestor::*[name()="a"])]')  # select anything that is not b or doesn't have an ancestor a
```
 please correct me if i'm wrong

Reply:
I think that's correct.